st2common/st2common/services/packs.py

```diff
@@ -26,3 +31,48 @@ def get_pack_by_ref(pack_ref):
     """
     pack_db = Pack.get_by_ref(pack_ref)
     return pack_db
+
+
+def fetch_pack_index(index_url=None):
+    """
+    Fetch the pack indexes (either from the config or provided as an argument)
+    and return the object.
+    """
+    if not index_url:
+        index_urls = cfg.CONF.content.index_url
+    elif isinstance(index_url, str):
+        index_urls = [index_url]
+    elif hasattr(index_url, '__iter__'):
+        index_urls = index_url
+    else:
+        raise TypeError('"index_url" should either be a string or an iterable object.')
+
+    result = {}
+    for index_url in index_urls:
+        result.update(requests.get(index_url).json())
+    return result
+
+
+def search_pack_index(query=None, pack=None):
+    """
+    Search the pack index either by pack name or by query.
+    Returns a pack object if the pack name is specified, otherwise returns
+    a list of matches for a query.
+    """
+    if (not query and not pack) or (query and pack):
```

Comment:
Yeah, but there's no xor operator in python (except for `^`), and this is the most explicit way that doesn't involve imports or weird casting :)